### ISSUE TITLE
V4.x Build `CloseCurrentDatabase2` closes AddIn and not in use database.

### ISSUE THREAD

**Issue by hecon5** [joyfullservice/msaccess-vcs-integration#356]
In my environment, running `CloseCurrentDatabase2` closes the addin window, and not the database when building a new addin from source.

This then causes an error when it hits the line `        Name strPath As strBackup` a few lines later, because it's not closed.

**Comment by joyfullservice** [comment]
@hecon5 - Are you referring to line 279 of `modImportExport` on the dev branch? Note that this code should be run from the add-in, not the current database. I am able to build the add-in without any problems, but I am also building v4 with v4.

I am attaching a copy of the latest build from the dev branch so you can see if this is still a problem. (I realize you might need to make an additional tweak to the dev version due to some specific environment considerations you have to work with.)
[Version Control_v4.04(dev).zip](https://github.com/joyfullservice/msaccess-vcs-integration/files/9584981/Version.Control_v4.04.dev.zip)

**Comment by hecon5** [comment]
Line 25 of `modWizHook.CloseCurrentDatabase2` is the line I mean. 

I'll run this and see if it works, one moment.

**Comment by joyfullservice** [comment]
Hmm... That's different than what happens on my system. 

Could you test something? Open the add-in database file, then open the add-in from the add-ins menu or the ribbon. At this point you should have both projects loaded in VBA.

![image](https://user-images.githubusercontent.com/7254992/190673665-a90910aa-c336-4acc-84ae-9d784da45a03.png)

Then, if you navigate to the add-in project (the ALL CAPS project), then run `CloseCurrentDatabase2` in the immediate window, which database closes? On my system it always closes the Current Database (the currently opened file, not the add-in). This is the case whether this code is run from the add-in project, or the current database project.

I realize that the WizHook functions are not officially documented or supported, but this one seems pretty self-explanatory, so I would be surprised if the behavior was changed to do something different in newer versions of Access. (But stranger things have happened.  😄)

**Comment by hecon5** [comment]
After running that, it closed the non-caps 
![image](https://user-images.githubusercontent.com/54177882/190676366-05e2845b-5f1d-4006-9bcd-db325faa3ad2.png)

**Comment by hecon5** [comment]
Interestingly, this worked correctly when I run it in isolation, but if I run build, it will not work.

**Comment by hecon5** [comment]
Both of these work correctly when run from Immediate Window, but if I open the addin Form and click build, it does not.

![image](https://user-images.githubusercontent.com/54177882/190677746-6cbaed78-9137-4f0a-bf18-0e317d38ac73.png)

**Comment by joyfullservice** [comment]
How are you opening the add-in form? Are you opening it from the add-ins menu? (This is important because you don't want to be running the form in the _current database_ you want to be running it in the _add-in database_ that is loaded at the application level.)

**Comment by hecon5** [comment]
Correct, Add ins menu; same as other databases which aren't the VCS.

![image](https://user-images.githubusercontent.com/54177882/190678748-4b2b201b-987f-4cf6-a1a9-124febf1ca30.png)

![image](https://user-images.githubusercontent.com/54177882/190678779-9f9cf252-3ce5-4fff-a842-25f72c1bb03f.png)

**Comment by joyfullservice** [comment]
Okay, thanks! I am able to reproduce the issue now. (It works fine when using the ribbon, which is why I hadn't seen this issue in my testing.)

**Comment by joyfullservice** [comment]
This turned out to be a pretty complex issue stemming from the management of object references. In a nutshell, when the build was initiated from the add-ins menu, the instance of the main form was still in the call stack of the build process, and could not be staged for loading in the main form when building the add-in from source. I resolved this by using an API timer callback to initiate the build, allowing the main form to be closed before kicking off the build (but still using the GUI).

I also cleaned up some other object reference issues with the staging process, so I can now successfully build the add-in and other projects from either the ribbon or the add-ins menu.

@hecon5 - Let me know if this resolves the build issues for you in relation to closing the current database.
